app/qml/main.qml

```diff
@@ -738,6 +738,11 @@ ApplicationWindow {
         target: __loader
         onLoadingStarted: projectLoadingScreen.visible = true
         onLoadingFinished: projectLoadingScreen.visible = false
+        onProjectReloaded: {
+          // clear all previous references to old project
```

Comment:
Can you please also add a short note about why we do it (e.g. "if we don't clear references to the previous project, highlights may end up with dangling pointers to map layers and cause crashes")